Deprecate recovery setup options in E2EE docs (#30681)
element-hq/element-web: docs/e2ee.md

Note: If the server is configured to forcibly enable encryption for some or all rooms,
this behaviour will be overridden.

# Setting up recovery

By default, Element strongly encourages (but does not require) users to set up

> recovery so that you can access history on your new devices as well as retain access to your message history and cryptographic identity when you lose all of your devices.

Reviewer: ```suggestion
recovery (previously known as Secure backup) so that they can retain access to their message history and cryptographic identity if they lose all of their devices.
```

## Deprecation notice

> The configuration options `secure_backup_required` and `secure_backup_setup_methods`
> in the `/.well-known/matrix/client` config have been deprecated, and are no longer applicable.

Reviewer: As techguy already said, though not in as many words: I don't think "deprecated, and are no longer applicable" makes sense. Either they are deprecated -- which implies that they still work for now, but maybe not for much longer -- or support has already been removed.

```suggestion
Support for the configuration options `secure_backup_required` and `secure_backup_setup_methods`
in the `/.well-known/matrix/client` config has been removed.
```

> The setup of the recovery is now always suggested to all users by showing a one off toast and a
> permanent red dot on the *Encryption* menu in the *Settings*. The methods on the UI are limited
> to the (generated) recovery key. Using a (custom) passphrase still works, but is not exposed on the
> UI when setting up recovery.

Reviewer: ```suggestion
Setting up recovery is now always recommended to all users by showing a one-off toast and a
permanent red dot on the *Encryption* tab in the *Settings* dialog. When creating a new
recovery key, the UI only supports auto-generated keys. Using an existing (custom) passphrase
still works, but is not exposed in the UI when setting up recovery.
```
